### DISCUSSION TITLE
Route file not automatically loaded

### DISCUSSION THREAD

**Discussion by tennox** [laravel/framework#34183] [General]
I read in the [docs](https://laravel.com/docs/7.x/routing#basic-routing) that:

> All Laravel routes are defined in your route files, which are located in the routes directory. These files are automatically loaded by the framework.

But just creating a new file `routes/foo.php` wasn't enough - it was never loaded.

I figured I needed to add it to **RouteServiceProvider** (by creating & calling a function that refers to the file via `->group(base_path('routes/web.php'))`).

Is it a mistake on my side or a documentation issue? (or both)

**Comment by Aferz** [comment]
Mmh @TeNNoX I think you misunderstood the docs about that topic.

The docs don't say __all__ your route files will be loaded by the framework. It means, once you create a new project, all the existing files in there will be loaded by the framework.

You are right about `RouteServiceProvider`. You just have to add them via that file: 

```php
    public function map()
    {
        // Framework routes
 
        $this->mapFooRoutes();
    }

    protected function mapFooRoutes()
    {
        Route::group(base_path('routes/foo.php'));
    }
```

I've never thought it could cause ambiguity but maybe because I'm an old user of Laravel. You could try submitting a PR to the docs with a better sentence that explains it clearly.

Cheers.
